sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -218,6 +218,11 @@ class SessionCatalog(
     if (dbName == DEFAULT_DATABASE) {
       throw new AnalysisException(s"Can not drop default database")
     }
+    if (cascade && databaseExists(dbName)) {
+      listTables(dbName).foreach { t =>
+        refreshTable(t)
```

Comment:
The goal of this PR is invalidating the tables. So, let's use more narrow API here.
```
-        refreshTable(t)
+        invalidateCachedTable(QualifiedTableName(dbName, t.table))
```